### ISSUE TITLE
Conserving STk

### ISSUE THREAD

**Issue by lassik** [egallesio/STklos#501]
We are slowly creating a comprehensive website at [Scheme.org](https://www.scheme.org/). As one part of the website, I've started a project known as [Scheme Conservatory](https://conservatory.scheme.org/). The idea is to preserve old Scheme implementations, archiving the websites, archiving old releases and writing simple patches to keep them running on modern operating systems.

I took the liberty of archiving the old STk website at [conservatory.scheme.org/stk](https://conservatory.scheme.org/stk/). I hope this is ok.

If there's interest from the current STklos maintainers, we could try to clean up the website (so that most of the links work, etc.) and try to build the last STk release on e.g. Linux or Mac.

What do you think?

**Comment by lassik** [comment]
Here is the git repo for the website: https://gitea.scheme.org/conservatory/stk-website

**Comment by jpellegrini** [comment]
That's cool!
Is there some way to file issues there? I'd like to suggest XLISP, whose latest version was close to R3RS.

**Comment by lassik** [comment]
Yes, it's possible to [register an account](https://gitea.scheme.org/user/sign_up). The user interface is almost identical to GitHub. For example, each repository has a similar Issues tab.

There is no official place to suggest new projects to conserve, but the [schemeorg mailing list](https://srfi.schemers.org/srfi-list-subscribe.html#schemeorg) is a reasonable place for the time being. Or email me privately.

If you could help with STk and XLISP, that would be great. I have some files for both, but probably not everything of interest.

**Comment by lassik** [comment]
Anyway, we should confirm with @egallesio that it's ok to go on conserving STk. (For many old projects the authors are hard to reach, so I added them without confirmation. But we are lucky that EG is still around.)

**Comment by egallesio** [comment]
> We are slowly creating a comprehensive website at [Scheme.org](https://www.scheme.org/). As one part of the website, I've started a project known as [Scheme Conservatory](https://conservatory.scheme.org/). The idea is to preserve old Scheme implementations, archiving the websites, archiving old releases and writing simple patches to keep them running on modern operating systems.
>

This is a really great idea! Thanks for the initiative.

> I took the liberty of archiving the old STk website at [conservatory.scheme.org/stk](https://conservatory.scheme.org/stk/). I hope this is ok.

Of course it is OK.

>
> If there's interest from the current STklos maintainers, we could try to clean up the website (so that most of the links work, etc.) and try to build the last STk release on e.g. Linux or Mac.
>
> What do you think?


That it is really awesome. It is a lot of work and I'm impressed by what has already be done (for instance with STk mailing list). If necessary, I have a copy of all the STk home page.

I have just tried to compile STk on Linux, and it still works (with a really *vintage* look for them GUI, but it works). I can of course help to maintain the Linux port, but it is really difficult for me to test t on a Mac (I can run MacOs on a VM, but I cannot easily launch it, and it is a world that I don't know).

**Comment by lassik** [comment]
Thanks!

> It is a lot of work and I'm impressed by what has already be done (for instance with STk mailing list).

Thankfully, the site was quite easy to obtain from the [Wayback Machine](https://web.archive.org/).

(For sites that are still online, another option would be `wget --mirror`.)

It looks like we got the whole mailing list archive, but I'm not sure. It would be nice to present the various Scheme mailing list archives in one place under https://lists.scheme.org/.

> If necessary, I have a copy of all the STk home page.

Can you send it as a tar file? I can diff it with the Wayback Machine copy to find out if there is anything missing.

BTW, do you have a version control repo of STk? If so, we could try to import it into git.

> I have just tried to compile STk on Linux, and it still works (with a really _vintage_ look for them GUI, but it works).

Good to hear! Did you have to patch anything, or did it run as-is?

> I can of course help to maintain the Linux port,

That would be terrific news. I assume not many changes are made to the basic Unix and Tk APIs anymore, so hopefully not much work will be needed.

> but it is really difficult for me to test t on a Mac (I can run MacOs on a VM, but I cannot easily launch it, and it is a world that I don't know).

I can test on Mac for the near future.

**Comment by egallesio** [comment]
> Can you send it as a tar file? I can diff it with the Wayback Machine copy to find out if there is anything missing.

Yes, I'll try to put it here as an attached file if it is not too big. Otherwise, we'll see...

> BTW, do you have a version control repo of STk? If so, we could try to import it into git.

The version I used for my test is just here on [GitHub](https://github.com/egallesio/STk). When I put it on GitHub, I applied some adaptations, that were sent to me,  for current versions of GCC. So, the git version is a bit different of the last released version.

> Did you have to patch anything, or did it run as-is?

I had just cloned the GitHub tree and compiled it!

> I can test on Mac for the near future.

That would be really great. Thanks.

**Comment by egallesio** [comment]
I tried to put it here as an attached file but the archive was too big (83Mb, and the limit is 25Mb). So, I have deleted all the releases (except the last one, in the source and binary directories). The tar archive has been embedded in a zip file, since Github is a bit picky on the allowed attached files. 


[STk-last.zip](https://github.com/egallesio/STklos/files/11076995/STk-last.zip)

**Comment by lassik** [comment]
Thanks!

To circumvent the size limit, can you make an `stk-website` repo and put all the files there? GitHub should be able to easily handle a 100 Mb repo.
